### ISSUE TITLE
dl.open error on windows

### ISSUE THREAD

**Issue by ninoles** [ronf/asyncssh#72]
I got the following error on import asyncssh in 1.8.0 on Windows (Python 3.5.2):
```
  File "E:\virtualenv\prototype\lib\site-packages\genvid\toolbox\terraform.py", line 11, in <module>           
    import asyncssh                                                                                            
  File "E:\virtualenv\prototype\lib\site-packages\asyncssh\__init__.py", line 23, in <module>                  
    from .agent import SSHAgentClient, SSHAgentKeyPair, connect_agent                                          
  File "E:\virtualenv\prototype\lib\site-packages\asyncssh\agent.py", line 42, in <module>                     
    from .public_key import SSHKeyPair                                                                         
  File "E:\virtualenv\prototype\lib\site-packages\asyncssh\public_key.py", line 28, in <module>                
    from .cipher import get_encryption_params, get_cipher                                                      
  File "E:\virtualenv\prototype\lib\site-packages\asyncssh\cipher.py", line 15, in <module>                    
    from .crypto import lookup_cipher                                                                          
  File "E:\virtualenv\prototype\lib\site-packages\asyncssh\crypto\__init__.py", line 38, in <module>           
    from .umac import umac32, umac64, umac96, umac128                                                          
  File "E:\virtualenv\prototype\lib\site-packages\asyncssh\crypto\umac.py", line 110, in <module>              
    _nettle = ctypes.cdll.LoadLibrary(ctypes.util.find_library('nettle'))                                      
  File "c:\program files\python35\Lib\ctypes\__init__.py", line 425, in LoadLibrary                            
    return self._dlltype(name)                                                                                 
  File "c:\program files\python35\Lib\ctypes\__init__.py", line 347, in __init__                               
    self._handle = _dlopen(self._name, mode)                                                                   
TypeError: bad argument type for built-in operation
```
Not sure the cause of it, but for sure, I don't nettle installed.

**Comment by ronf** [comment]
Hmm. It seems that the ctypes.cdll.LoadLibrary() function on UNIX systems has no problem coping with a return value of None from ctypes.util.find_library(), but on Windows this throws a TypeError. The calling code in AsyncSSH is prepared to receive ImportError, AttributeError, and OSError but not TypeError. Sorry I didn't catch this -- I don't have my unit tests running on Windows yet.

The fix should be fairly simple. In crypto/__init__.py, change:

```python
# Import umac cryptographic hash if available
try:
    from .umac import umac32, umac64, umac96, umac128
except (ImportError, AttributeError, OSError): # pragma: no cover
    pass
```

to:

```python
# Import umac cryptographic hash if available
try:
    from .umac import umac32, umac64, umac96, umac128
except (ImportError, AttributeError, OSError, TypeError): # pragma: no cover
    pass
```

I'll try to get a fix out for this quickly.

**Comment by ronf** [comment]
After looking at this more closely, I went with a slightly different fix. I also confirmed that the UMAC support works on Windows, if the Nettle library and necessary dependencies are installed.

These changes are released in AsyncSSH version 1.8.1 which should now be available on PyPI.
